### PR TITLE
Hilbert curve

### PR DESCRIPTION
**Pull request by villares**
Java & Python

### CODE REVIEW

File: examples/Java/HilbertCurve/HilbertCurve.pde [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      line(0, 0, 0, -strokeLen);
+      translate(0, -strokeLen);
+    } else if (c == '+') {
+      rotate(angle);
+    } else if (c == '-') {
+      rotate(-angle);
+    } else if (c == '[') {
+      pushMatrix();
+    } else if (c == ']') {
+      popMatrix();
```
Comment (jeremydouglass) [inline examples/Java/HilbertCurve/HilbertCurve.pde]: Is this stack feature part of a generalized L-system implementation? It isn't used by the sketch. Same in the .pyde.

File: examples/Java/HilbertCurve/HilbertCurve.pde [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    angleDeg -= 1;
+    println("Angle: " + angleDeg);
+  }
+  if (key == '=' || key == '+') {
+    angleDeg += 1;
+    println("Angle: " + angleDeg);
+  }
+  if (key == 's') {
+    saveFrame("####.png");
+  }
```
Comment (jeremydouglass) [inline examples/Java/HilbertCurve/HilbertCurve.pde]: On the wiki I would say this starts to feel like feature creep -- the code is supposed to demonstrate one task, not be a full-featured application. But the navigating and screenshot features are pretty cool, and they don't complicate the rest of the code too much so I figure it is fine to leave them.

File: examples/Python/HilbertCurve/HilbertCurve.pyde [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            translate(0, -stroke_len)
+            # ellipse(0, 0, 10, 10)
+        elif c == "+":
+            rotate(angle)
+        elif c == "-":
+            rotate(-angle)
+        elif c == "[":
+            pushMatrix()
+        elif c == "]":
+            popMatrix()
```
Comment (jeremydouglass) [inline examples/Python/HilbertCurve/HilbertCurve.pyde]: Same as previous file

File: examples/Python/HilbertCurve/HilbertCurve.pyde [hunk 4]
```diff
@@ -0,0 +1,84 @@
+"""
+Hilbert Curve
+https://www.rosettacode.org/wiki/Hilbert_curve#Processing
+Processing 3.4
+2020-03-25 Alexandre Villares
+
+Task:
+
+Produce a graphical or ASCII-art representation of a Hilbert curve of at least order 3.
```
Comment (jeremydouglass) [inline examples/Python/HilbertCurve/HilbertCurve.pyde]: wrap to 80

File: examples/Java/HilbertCurve/HilbertCurve.pde [hunk 5]
```diff
@@ -0,0 +1,104 @@
+/**
+ * Hilbert Curve
+ * https://www.rosettacode.org/wiki/Hilbert_curve#Processing
+ * Processing 3.4
+ * 2020-03-25 Alexandre Villares
+ * 
+ * Task:
+ * 
+ * Produce a graphical or ASCII-art representation of a Hilbert curve of at least order 3.
```
Comment (jeremydouglass) [inline examples/Java/HilbertCurve/HilbertCurve.pde]: wrap to 80

**Review by jeremydouglass — CHANGES_REQUESTED** [review]
Fantastic. Very minor revision and I'll merge.

File: examples/Java/HilbertCurve/HilbertCurve.pde (hunk 1 above)
Comment (villares) [reply examples/Java/HilbertCurve/HilbertCurve.pde]: Yeah! I'll remove this!

**Comment by villares** [comment]
I guess I'm done now :)

File: examples/Java/HilbertCurve/HilbertCurve.pde (hunk 1 above)
Comment (jeremydouglass) [reply examples/Java/HilbertCurve/HilbertCurve.pde]: Thank you!

**Pull request merged**